sample/TKTransferSamples.m

```diff
@@ -0,0 +1,215 @@
+    
+    [self runUntilTrue:^ {
+        return waitingForPayment;
+    }];
+}
+
+- (void)testCreateEndorseTransferToken {
+    TokenIOSync *tokenIOSync = [[self sdkBuilder] buildSync];
+    TKMemberSync *payerSync = [self createMember:tokenIOSync];
+    TKMember *payer = payerSync.async;
+    TKAccountSync *payerAccountSync = [payerSync linkAccounts:[self createBankAuthorization:payerSync]][0];
+    TKAccount *payerAccount = payerAccountSync.async;
+    Alias *payeeAlias = [self generateAlias];
+    TKMemberSync *payeeSync = [tokenIOSync createMember:payeeAlias];
+    TKMember *payee = payeeSync.async;
+    NSString *refId = @"purchase:2017-11-01:28293336394ffby";
+    __block Token *transferToken = nil;
+    
+    // createTransferToken begin snippet to include in docs
+    TransferTokenBuilder *builder = [payer createTransferToken:100.0
+                                                      currency:@"EUR"];
+    builder.accountId = payerAccount.id;
+    builder.redeemerAlias = payeeAlias;
+    builder.descr = @"Book purchase";
+    builder.refId = refId;
+    
+    [builder executeAsync:^(Token *t){
+        // Use token.
+        transferToken = t;
+    }   onError:^(NSError *e) {
+        // Something went wrong. (We don't just build a structure; we also
+        // upload it to Token cloud. So things can go wrong.)
+        @throw [NSException exceptionWithName:@"BuilderExecuteException"
+                                       reason:[e localizedFailureReason]
+                                     userInfo:[e userInfo]];
+    }];
+    // createTransferToken done snippet to include in docs
+    
+    [self runUntilTrue:^ {
+        return (transferToken != nil);
+    }];
+    
+    // endorseToken begin snippet to include in docs
+    [payer endorseToken:transferToken
+                withKey:Key_Level_Standard
+              onSuccess:^(TokenOperationResult *result) {
+                  // Update transferToken with newer value:
+                  // Payload is same; now has signatures attached.
+                  transferToken = result.token;
+              } onError:^(NSError *e) {
+                  // something went wrong
+                  @throw [NSException exceptionWithName:@"EndorseException"
+                                                 reason:[e localizedFailureReason]
+                                               userInfo:[e userInfo]];
+              }];
+    // endorseToken done snippet to include in docs
+    
+    [self runUntilTrue:^ {
+        return (transferToken.payloadSignaturesArray_Count > 0);
+    }];
+    
+    __block Transfer *transfer = nil;
+    
+    // redeemToken begin snippet to include in docs
+    TransferEndpoint *destination = [[TransferEndpoint alloc] init];
+    // There are a few ways to specify destination; here we use (fake) IBAN
+    destination.account.sepa.iban = @"123";
+    [payee getToken:transferToken.id_p
+          onSuccess:^(Token *token) {
```

Comment:
Do you want to include the usage of get token here? Otherwise you can just redeem the transfertoken object

Reply:
Yeah. The flow in this sample has payee doing the redeem. Payee's stepup notification has an id.

Reply:
"Payee's stepup notification has an id." I was wrong here. We don't have a notification where we expect the notified-person to redeem. 

Hmm, the flow in our sample code has always been "payee redeems". The story has a web merchant and a shopper. And you're thinking "Do we expect a web merchants to run a web server on an iPhone?" and ... no... no we don't. D'oh. 

I just created a JIRA. I can clean this up.